bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,450 @@
+NB: When authentication is enabled, the environment variable ```JGROUPS_CLUSTER_PASSWORD``` must be set.
+
+Some examples:
+
+Configure for OpenShift execution:
+
+```
+<cloud/>
+```
+
+Configure for OpenShift execution with jgroups authentication enabled:
+
+```
+<cloud>
+  <enable-jgroups-password>true</enable-jgroups-password>
+</cloud>
+```
+Configure for kubernetes execution:
+
+```
+<cloud>
+  <type>kubernetes</type>
+</cloud>
+```
+
+===== WildFly OpenShift operator
+
+The WildFly OpenShift operator can be used to manage deployments based on image containing a WildFly bootable JAR.
+At boot time, the WildFly bootable JAR dumps in the file ```/opt/jboss/container/wildfly-bootable-jar/install-dir``` its installation path.
+This information is required by the WildFly OpenShift operator to retrieve transaction logs and call into WildFly CLI.
+
+===== JKube Maven plugin
+
+The JKube Maven Plugin is evolved with a generator that recognizes the bootable JAR maven plugin. DETAILS TO BE PROVIDED WHEN FINALIZED.
+
+=== Nice-to-Have Requirements
+
+* Add the ability to copy content in the server during build. That is a common requirement (e.g.: auth properties files).
+* Ability to generate a runtime Maven repository in order to resolve the server artifacts from Maven local cache. 
+Huge benefit in term of JAR size and boot time (4/5 time faster, around 160ms to start vs 700 ms). 
+Specifically in a docker/openshift context, relying on slim server + Maven repository speeds up startup without impacting image size.
+Eg: java -Dmaven.repo.local=/maven-repo -jar myapp-bootable.jar
+
+
+=== Non-Requirements
+
+* Offer a new way to configure the server (e.g.: Thorntail yaml file).
+* Package a custom standalone XML file. Standalone XML file is generated during build by the Maven plugin and can't be replaced. 
+* Usage of a different log manager than JBoss log manager is out of scope.
+* No domain support.
+
+== Implementation Plan
+
+* Evolve wildfly-core with a runtime to boot the bootable JAR.
+* Develop new Maven plugin.
+
+== Test Plan
+
+* Maven plugin https://github.com/wildfly-extras/wildfly-jar-maven-plugin/tree/master/tests[tests] (in plugin repo)
+* Bootable runtime https://github.com/wildfly/wildfly-core/tree/master/bootable-jar/runtime/src/test/java/org/wildfly/core/jar/runtime[tests] (in wildfly-core repo)
+* Wildfly-core https://github.com/wildfly/wildfly-core/blob/master/testsuite/pom.xml#L425[tests] (in wildfly-core repo). 
+Run existing tests (when applicable) against bootable JAR using the _-Dts.bootable_ maven profile: ```cd <wildfly repo>/testsuite; mvn clean install -Dts.bootable```
+
+== Community Documentation
+
+* https://github.com/wildfly-extras/wildfly-jar-maven-plugin/releases/download/2.0.0.Alpha4/index.html[Maven plugin documentation]
```

Comment:
Isn't this (and below) `Alpha6`at the moment?

Reply:
Will be fixed with beta1 ASAP.

Reply:
Ok, thanks